### DISCUSSION TITLE
What is the general type for ctypes.Array and np.ndarray?

### DISCUSSION THREAD

**Discussion by monchin** [python/typing#1290] [Q&A] [answered]
I have a function, and one parameter of it is an array, but it could be either `multiprocessing.sharedctypes.RawArray`(with typing `ctypes.Array`) or `numpy.ndarray`. In my opinion, both of these could be describe as `array`, but I can't find a type to contain both. I tried `Sequence` and `Iterable`, neither contains `RawArray`. So is there any suggestions? Or the only way to solve it is `Union`?

**Comment by hauntsaninja** [comment, marked as answer]
A Union is very likely what you want. You could also potentially write a Protocol defining exactly what you expect the shared behaviour to be.
